lib/ex_machina.ex

```diff
@@ -73,21 +73,12 @@ defmodule ExMachina do
     end
   end
 
-  defmacro factory(factory_name, do: block) do
```

Comment:
I'm wondering if rather than removing this we should add raise a helpful error on the `factory` and `assoc` macros. Just for this version so that it makes an upgrade easy. I want the upgrade process to be as smooth as possible. A lot of people upgrade without looking at release notes so I think something like this would help:

```elixir
# I haven't tested this so it may not work exactly as-is
defmacro factory(factory_name, _attrs) do
  raise """
  The factory macro has been removed. Please use a regular function instead.

    def factory(#{factory_name}, attrs) do
        ...
    end
  """
end
```